addons/portal/views/portal_templates.xml

```diff
@@ -516,30 +520,45 @@
         </div>
         <div t-attf-class="mb-3 #{error.get('country_id') and 'o_has_error' or ''} col-xl-6">
             <label class="col-form-label" for="country_id">Country</label>
-            <select name="country_id" t-attf-class="form-select #{error.get('country_id') and 'is-invalid' or ''}" t-att-disabled="None if partner_can_edit_vat else '1'">
+            <select name="country_id" t-attf-class="form-select #{error.get('country_id') and 'is-invalid' or ''}" t-att-disabled="None if not partner_can_edit_company else '1'">
                 <option value="">Country...</option>
                 <t t-foreach="countries or []" t-as="country">
                     <option t-att-value="country.id" t-att-selected="country.id == int(country_id) if country_id else country.id == partner.country_id.id">
                         <t t-esc="country.name" />
                     </option>
                 </t>
             </select>
-            <small t-if="not partner_can_edit_vat" class="form-text text-muted d-block d-xl-none">Changing the country is not allowed once document(s) have been issued for your account. Please contact us directly for this operation.</small>
+            <small t-if="partner_can_edit_company" class="form-text text-muted d-block d-xl-none">Changing the country is not allowed once document(s) have been issued for your account. Please contact us directly for this operation.</small>
+        </div>
+        <div t-attf-class="col-xl-6 mb-3" t-att-hidden="true">
+            <label class="col-form-label label-optional" for="type">Use this address for</label>
+            <select name="type" t-attf-class="form-select">
+                <option value="other" t-att-selected="type == 'other' or partner.type == 'other'"></option>
+                <option value="delivery" t-att-selected="type == 'delivery' or partner.type == 'delivery'">Delivery</option>
+                <option value="invoice" t-att-selected="type == 'invoice' or partner.type == 'invoice'">Invoice</option>
+                <option value="contact" t-att-selected="type == 'contact' or partner.type == 'contact'">Contact</option>
+            </select>
         </div>
     </template>
 
     <template id="portal_my_details">
         <t t-call="portal.portal_layout">
             <t t-set="additional_title">Contact Details</t>
-            <form action="/my/account" method="post">
+            <t t-if="add_address">
+                <t t-set="new_address_href" t-valuef="/my/account/save?address={{ add_address }}"/>
+            </t>
+            <t t-else="">
+                <t t-set="new_address_href" t-valuef="/my/account/save?address={{ add_address }}"/>
+            </t>
+            <form t-att-action="new_address_href" method="post">
```

Comment:
If value is same in either condition we should directly use it in form action 
```suggestion
            <form t-att-action="/my/account/save?address={{ add_address }}f" method="post">
 ```
 I haven't check the code please adjust it accordingly